### PR TITLE
en: add fields in br crd

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added, or deleted? (Required)

Add missing fields to backup restore CR.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/1738
- Other reference link(s):

### CODE REVIEW

File: en/backup-restore-cr.md [hunk 1]
```diff
@@ -124,8 +139,10 @@ This section introduces the fields in the `Backup` CR.
 * `spec.s3.region`: if you want to use Amazon S3 for backup storage, configure this field as the region where Amazon S3 is located.
 * `.spec.s3.bucket`: the name of the bucket of the S3-compatible storage.
 * `.spec.s3.prefix`: if you set this field, the value is used to make up the remote storage path `s3://${.spec.s3.bucket}/${.spec.s3.prefix}/backupName`.
+* `.spec.s3.path`: specifies the storage path of the backup file on the remote storage. This field is valid only when using Dumpling to back up data or using TiDB Lightning to restore data. For example, `s3://test1-demo1/backup-2019-12-11T04:32:12Z.tgz`.
```
Comment (qiancai) [inline en/backup-restore-cr.md:142]:
```suggestion
* `.spec.s3.path`: specifies the storage path of backup files on the remote storage. This field is valid only when the data is backed up using Dumpling or restored using TiDB Lightning. For example, `s3://test1-demo1/backup-2019-12-11T04:32:12Z.tgz`.
```

File: en/backup-restore-cr.md [hunk 2]
```diff
@@ -155,9 +172,12 @@ This section introduces the fields in the `Backup` CR.
 ### GCS fields
 
 * `.spec.gcs.projectId`: the unique identifier of the user project on GCP. To obtain the project ID, refer to [GCP documentation](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+* `.spec.gcs.location`: the location of the GCS bucket. For example, `us-west2`.
+* `.spec.gcs.path`: the storage path of the backup file on the remote storage. This field is valid only when using Dumpling to back up data or using TiDB Lightning to restore data. For example, `gcs://test1-demo1/backup-2019-11-11T16:06:05Z.tgz`.
```
Comment (qiancai) [inline en/backup-restore-cr.md:176]:
```suggestion
* `.spec.gcs.path`: the storage path of backup files on the remote storage. This field is valid only when the data is backed up using Dumpling or restored using TiDB Dumpling. For example, `gcs://test1-demo1/backup-2019-11-11T16:06:05Z.tgz`.

File: en/backup-restore-cr.md [hunk 3]
```diff
@@ -155,9 +172,12 @@ This section introduces the fields in the `Backup` CR.
...
+* `.spec.gcs.secretName`: the name of the secret which stores the GCS account credential.
```
Comment (qiancai) [inline en/backup-restore-cr.md:177]:
```suggestion
* `.spec.gcs.secretName`: the name of the secret that stores the GCS account credential.
```

File: en/backup-restore-cr.md [hunk 4]
```diff
@@ -212,6 +232,29 @@ This section introduces the fields in the `Restore` CR.
     - When using BR for restoring, you can specify the BR version in this field. For example,`spec.toolImage: pingcap/br:v5.4.0`. If not specified, `pingcap/br:${tikv_version}` is used for restoring by default.
     - When using Lightning for restoring, you can specify the Lightning version in this field. For example, `spec.toolImage: pingcap/lightning:v5.4.0`. If not specified, the Lightning version specified in `TOOLKIT_VERSION` of the [Backup Manager Dockerfile](https://github.com/pingcap/tidb-operator/blob/master/images/tidb-backup-manager/Dockerfile) is used for restoring by default.
 
+* `.spec.backupType`: the restore type. This field is valid only when using BR to restore data. Currently, the following three types are supported, and this field can be combined with the `.spec.tableFilter` field to configure table filter rules:
+    * `full`: restore all databases in the TiDB cluster.
+    * `db`: restore one database in the TiDB cluster.
+    * `table`: restore one table in the TiDB cluster.
+
```
Comment (qiancai) [inline en/backup-restore-cr.md:239]: ditto

File: en/backup-restore-cr.md [hunk 5]
```diff
@@ -212,6 +232,29 @@ This section introduces the fields in the `Restore` CR.
...
+* `.spec.tikvGCLifeTime`: The temporary `tikv_gc_life_time` time setting during the restore, which defaults to `72h`.
```
Comment (qiancai) [inline en/backup-restore-cr.md:240]:
```suggestion
* `.spec.tikvGCLifeTime`: the temporary `tikv_gc_life_time` setting during the restore, which defaults to `72h`.
```

File: en/backup-restore-cr.md [hunk 6]
```diff
@@ -212,6 +232,29 @@ This section introduces the fields in the `Restore` CR.
...
+
+    Before the restore begins, if the `tikv_gc_life_time` setting in the TiDB cluster is smaller than `spec.tikvGCLifeTime` set by the user, TiDB Operator [adjusts the value of `tikv_gc_life_time`](https://docs.pingcap.com/tidb/stable/dumpling-overview#tidb-gc-settings-when-exporting-a-large-volume-of-data) to the value of `spec.tikvGCLifeTime`. This operation makes sure that the restored data is not garbage-collected by TiKV.
```
Comment (qiancai) [inline en/backup-restore-cr.md:242]:
```suggestion
    Before the restore begins, if the `tikv_gc_life_time` setting in the TiDB cluster is smaller than `spec.tikvGCLifeTime` set by users, TiDB Operator [adjusts the value of `tikv_gc_life_time`](https://docs.pingcap.com/tidb/stable/dumpling-overview#tidb-gc-settings-when-exporting-a-large-volume-of-data) to the value of `spec.tikvGCLifeTime`. This operation makes sure that the restored data is not garbage-collected by TiKV.
```

File: en/backup-restore-cr.md [hunk 7]
```diff
@@ -212,6 +232,29 @@ This section introduces the fields in the `Restore` CR.
...
+
+    After the restore, whether the restore is successful or not, as long as the previous `tikv_gc_life_time` value is smaller than `.spec.tikvGCLifeTime`, TiDB Operator tries to set `tikv_gc_life_time` to the previous value.
```
Comment (qiancai) [inline en/backup-restore-cr.md:244]:
```suggestion
    After the restore, whether the restore is successful or not, as long as the original `tikv_gc_life_time` value is smaller than `.spec.tikvGCLifeTime`, TiDB Operator tries to set `tikv_gc_life_time` back to the original value.
```

**Review by qiancai — COMMENTED** [review]
@ran-huang I've completed the review and added some comments. PTAL. Thanks.

File: en/backup-restore-cr.md [hunk 8]
```diff
@@ -212,6 +232,29 @@ This section introduces the fields in the `Restore` CR.
     - When using BR for restoring, you can specify the BR version in this field. For example,`spec.toolImage: pingcap/br:v5.4.0`. If not specified, `pingcap/br:${tikv_version}` is used for restoring by default.
     - When using Lightning for restoring, you can specify the Lightning version in this field. For example, `spec.toolImage: pingcap/lightning:v5.4.0`. If not specified, the Lightning version specified in `TOOLKIT_VERSION` of the [Backup Manager Dockerfile](https://github.com/pingcap/tidb-operator/blob/master/images/tidb-backup-manager/Dockerfile) is used for restoring by default.
 
+* `.spec.backupType`: the restore type. This field is valid only when using BR to restore data. Currently, the following three types are supported, and this field can be combined with the `.spec.tableFilter` field to configure table filter rules:
+    * `full`: restore all databases in the TiDB cluster.
+    * `db`: restore one database in the TiDB cluster.
+    * `table`: restore one table in the TiDB cluster.
```
Comment (ran-huang) [inline en/backup-restore-cr.md:238]:
```suggestion
* `.spec.backupType`: the restore type. This field is valid only when you use BR to restore data. Currently, the following three types are supported, and this field can be combined with the `.spec.tableFilter` field to configure table filter rules:
    * `full`: restore all databases in a TiDB cluster.
    * `db`: restore a specifed database in a TiDB cluster.
    * `table`: restore a specified table in a TiDB cluster.
```

Comment (ran-huang) [reply en/backup-restore-cr.md:238]: @qiancai PTAL. Thanks!

Comment (qiancai) [reply en/backup-restore-cr.md:238]: LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**